packages/excalidraw/element/linearElementEditor.ts

```diff
@@ -279,10 +279,31 @@ export class LinearElementEditor {
         selectedPointsIndices.length === 1 &&
         element.points.length > 1
       ) {
-        const selectedIndex = selectedPointsIndices[0];
-        const referencePoint =
-          element.points[selectedIndex === 0 ? 1 : selectedIndex - 1];
+        let selectedIndex = selectedPointsIndices[0];
+        
+        // For elbow arrows with shift pressed
+        if (isElbowArrow(element)) {
+          const deltaX = scenePointerX - (element.x + element.points[selectedIndex === 0 ? 1 : selectedIndex - 1][0]);
+          const deltaY = scenePointerY - (element.y + element.points[selectedIndex === 0 ? 1 : selectedIndex - 1][1]);
+          
+          // Calculate angle in degrees (0-360)
+          const angle = ((Math.atan2(deltaY, deltaX) * 180) / Math.PI + 360) % 360;
+          const snapAngle = Math.round(angle / 15) * 15;
+          // Only remove midpoints for cardinal angles
+          if ([0, 90, 180, 270, 360].includes(snapAngle)) {
+            const newPoints = [
+              element.points[0],
+              element.points[element.points.length - 1]
+            ];
+            mutateElement(element, { points: newPoints });
```

Comment:
Why are you not using `mutateElbowArrow()` which can recalculate the arrow if needed? If you directly mutate an elbow arrow it might break the 90 degree angles if it is not adjusted (i.e. it will "jump" the next time somebody drag it).